src/functions/operatorname.js

```diff
@@ -30,37 +30,44 @@ defineFunction({
             let mode = "";
 
             // Consolidate Greek letter function names into symbol characters.
-            const temp = html.buildExpression(group.value.value, options, true);
+            const temp = html.buildExpression(
+                group.value.value, options.withFontFamily("mathrm"), true);
 
             // All we want from temp are the letters. With them, we'll
             // create a text operator similar to \tan or \cos.
             for (let i = 0; i < temp.length; i++) {
-                letter = temp[i].value;
+                const child = temp[i];
 
-                // In the amsopn package, \newmcodes@ changes four
-                // characters, *-/:’, from math operators back into text.
-                // Given what is in temp, we have to address two of them.
-                letter = letter.replace(/\u2212/, "-");   // minus => hyphen
-                letter = letter.replace(/\u2217/, "*");
+                if (child instanceof domTree.symbolNode) {
+                    letter = child.value;
 
-                // Use math mode for Greek letters
-                mode = (/[\u0391-\u03D7]/.test(letter) ? "math" : "text");
-                output.push(buildCommon.mathsym(letter, mode));
+                    // In the amsopn package, \newmcodes@ changes four
+                    // characters, *-/:’, from math operators back into text.
+                    // Given what is in temp, we have to address two of them.
+                    letter = letter.replace(/\u2212/, "-");   // minus => hyphen
+                    letter = letter.replace(/\u2217/, "*");
+
+                    // Use math mode for Greek letters
+                    mode = (/[\u0391-\u03D7]/.test(letter) ? "math" : "text");
+                    output.push(buildCommon.mathsym(letter, mode));
+                } else {
+                    output.push(child);
+                }
             }
         }
-        return buildCommon.makeSpan(["mop"], output, options);
+        const span = buildCommon.makeSpan(["mop"], output, options);
+        return span;
```

Comment:
(optional) Any particular reason this was split into 2 lines? For debugging perhaps?